### ISSUE TITLE
White screen browser doesn't close during authorizeAndExchangeCode

### ISSUE THREAD

**Issue by JKornberg** [MaikuB/flutter_appauth#231]
Platform: Google Pixel 4
Flutter Version: beta 2.5.0-5.1
Package version: ^1.1.0

I've successfully reached the sign in screen, signed up an account that shows up (in my case using Azure B2C Active Directory), but afterwards the browser window does not close. Subsequent tries show a blank window (white screen which I assume is authenticated) but does not redirect. I am only testing on Android

This is a demo app so i don't mind sharing the auth info:

```
AuthorizationTokenRequest('fb105d3b-53c3-47fa-8e95-e7ec1bced50f',
            'com.example.azureflutter://oauthredirect',
            discoveryUrl:
                'https://felasco.b2clogin.com/felasco.onmicrosoft.com/v2.0/.well-known/openid-configuration?p=B2C_1_azureflutter_signin_signup',
            scopes: ['openid', 'profile', 'email', 'offline_access', 'api']),
      );
```

My app package is: "com.example.azureflutter"

My redirect URL on azure is: "com.example.azureflutter://oauthredirect"

 Any advice would be greatly appreciated.

**Comment by MaikuB** [comment]
A browser not closing typically means you have a configuration issue with the redirect URL. This might be a server-side configuration issue or that you haven't done all of the setup properly on the client side e.g. the platform-specific configuration mentioned in the readme. You'd need to double check what you've done and compare it with the relevant docs and code samples. There's an example app for the plugin within the repository. Even though it's to connect with IdentityServer, the setup required is independent of what you use as your IdP

**Comment by JKornberg** [comment]
> A browser not closing typically means you have a configuration issue with the redirect URL. This might be a server-side configuration issue or that you haven't done all of the setup properly on the client side e.g. the platform-specific configuration mentioned in the readme. You'd need to double check what you've done and compare it with the relevant docs and code samples. There's an example app for the plugin within the repository. Even though it's to connect with IdentityServer, the setup required is independent of what you use as your IdP

Just to clarify, would an issue with the redirect url would cause the future to not complete (it is stuck on the await)? I will look into this today but I want to make sure this is the problem

**Comment by JKornberg** [comment]
2 concerns:
1. I wasn't able to enable AndroidX as described in the  ReadMe, but I read that for newer versions of Flutter, it is automatically enabled.
2. On the Azure end, I have a redirect URI set to: com.example.azureflutter://oauthredirect, does that seem correct?

**Comment by JKornberg** [comment]
wooo just got it working. In case anyone stumbles upon this, I was following this guide: https://medium.com/flutter-community/flutter-azure-authentication-with-ad-b2c-8b76c81dd48e

What I didn't realize is I thought the custom URL redirect should be the package url (in my case com.example.azureflutter). Instead it should be a custom one (I switched to com.example.azureplatform). This change allowed the Future to return and gave me a few more errors that I had to fix, like enabling "public client flows" within Azure.
